pkg/util/kubernetes/autoscalers/datadogexternal.go

```diff
@@ -61,6 +65,12 @@ const (
 	queryEndpoint = "/api/v1/query"
 )
 
+var (
+	refreshPeriod  = config.Datadog.GetInt("external_metrics_provider.refresh_period")
```

Comment:
I don't remember if `config.Datadog` is already initialized when the `var` is created. Did you get the chance to test this ?

Reply:
Yes I have tested this by building a custom cluster agent image and deploying it on a cluster with several hpa and DatadogMetric objects. The output shows that the refresh period is respected. So I think `refreshPeriod` is correctly initialized.

Reply:
This is correct, `config.Datadog` will not be parsed yet, you will always get the default value. You will probably need a lazy instantiation on first call with a `sync.Once`.